biothings/utils/hub.py

```diff
@@ -858,52 +762,57 @@ def __init__(self, paths, reload_func, wait=5):
         self.mod.add_reload_hook(self.reload_func)
         # only listen to these events. Note: directory detection is done via a flag so
         # no need to use IS_DIR
-        self.paths = []  # cleaned
-        for path in paths:
-            if not os.path.isabs(path):
-                path = os.path.abspath(path)
-            self.paths.append(path)
-            self.mod.watch(path)
-            for dirpath, dirnames, filenames in os.walk(path):
-                if exclude_from_reloader(dirpath):
-                    continue
-                if not dirnames:
-                    # reaching files (leaves)
-                    for fn in filenames:
-                        f_path = os.path.join(dirpath, fn)
-                        if exclude_from_reloader(f_path):
-                            continue
-                        self.mod.watch(f_path)
+        self.add_watch(paths)
+
         self.wait = wait
-        self.do_monitor = False
 
     def monitor(self):
         logging.info(
             "Monitoring source code in, %s:\n%s", repr(self.paths), pformat(self.watched_files())
         )
         self.mod.start(self.wait * 1000)  # millis
 
+    def add_watch(self, paths):
+        input_paths = paths.copy()
```

Comment:
add a doc_string @sengineer0

Reply:
@newgene 
It's added. Please have a check.